service/orchestrator/certificates.go

```diff
@@ -0,0 +1,172 @@
+	if !areAllAllowed && !slices.Contains(allowedCloudServices, res.CloudServiceId) {
+		// Important to nil the response since it is set already
+		return nil, status.Error(codes.PermissionDenied, service.ErrPermissionDenied.Error())
+	}
+
+	return
+}
+
+// ListCertificates implements method for getting a certificate, e.g. to show its state in the UI
+func (svc *Service) ListCertificates(ctx context.Context, req *orchestrator.ListCertificatesRequest) (res *orchestrator.ListCertificatesResponse, err error) {
+	// Validate request
+	err = service.ValidateRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// We only list certificates the user is authorized to see (w.r.t. the cloud service)
+	var conds []any
+	areAllAllowed, cloudServices := svc.authz.AllowedCloudServices(ctx)
+	if !areAllAllowed {
+		conds = append([]any{"cloud_service_id IN ?"}, []any{cloudServices})
+	}
+
+	res = new(orchestrator.ListCertificatesResponse)
+
+	res.Certificates, res.NextPageToken, err = service.PaginateStorage[*orchestrator.Certificate](req, svc.storage,
+		service.DefaultPaginationOpts, conds...)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not paginate results: %v", err)
+	}
+
+	return
+}
+
+// UpdateCertificate implements method for updating an existing certificate
+// Todo: Add auth
+func (svc *Service) UpdateCertificate(ctx context.Context, req *orchestrator.UpdateCertificateRequest) (response *orchestrator.Certificate, err error) {
+	// Validate request
+	err = service.ValidateRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check authorization
+	if !svc.authz.CheckAccess(ctx, service.AccessUpdate, req) {
+		err = service.ErrPermissionDenied
+		return
+	}
+
+	count, err := svc.storage.Count(req.Certificate, req.Certificate.Id)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "database error: %v", err)
+	}
+
+	if count == 0 {
+		return nil, status.Error(codes.NotFound, certificationNotFoundErrorMessage)
+	}
+
+	response = req.Certificate
+
+	err = svc.storage.Save(response, "Id = ?", response.Id)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "database error: %v", err)
+	}
+
+	logging.LogRequest(log, logrus.DebugLevel, logging.Update, req)
+
+	return
+}
+
+// RemoveCertificate implements method for removing a certificate
+func (svc *Service) RemoveCertificate(ctx context.Context, req *orchestrator.RemoveCertificateRequest) (response *emptypb.Empty, err error) {
+	// Validate request
+	err = service.ValidateRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Only remove certificate if user is authorized for the corresponding cloud service
+	areAllAllowed, allowedCloudService := svc.authz.AllowedCloudServices(ctx)
+	// Todo(lebogg to all): Is this too much? A user having admin flag off but no cloud services attached would have full access otherwise. Would mean we have to change other endpoints with auth as well, I guess.
```

Comment:
I am confused. Why should this be? If one has "all" off and the list of cloud services is empty, then the query will be `and cloud_service_id IN ?`, {}; this will always be empty so no certificates should be returned?